### PR TITLE
Boyscouting and more testing

### PR DESCRIPTION
**Pull request by friedger**
This PR removes unused files and adapts to some naming conventions.

In addition, it adds some more testing. The evaluation of the gradle project for testing has been moved into `BeforeClass`, different aspects of the tests have been separated into their own methods.

The tests run successful. 

However, the BintrayUploadTask is not configured correctly before execution, hence the test results for this method don't apply to the use gradle build cycle.

### CODE REVIEW

File: core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy [hunk 1]
```diff
@@ -73,9 +81,21 @@ class TestBintrayRelease {
...
+    public void testBintrayUploadTask() {
+        Task task = project.getTasks().findByPath(":bintrayUpload")
+        assertThat(task).isNotNull()
+
+        BintrayUploadTask uploadTask = task as BintrayUploadTask
+        uploadTask.user = "user"
+        uploadTask.apiKey = "apiKey"
 
-        // uploadTask verification here
+        uploadTask.repoName = "maven"
```
Comment (friedger) [inline core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy]: `repoName` and `packageName` are usually set after all projects have been evaluated. This callback (https://github.com/bintray/gradle-bintray-plugin/blob/master/src/main/groovy/com/jfrog/bintray/gradle/BintrayPlugin.groovy#L30) is not executed during test.

File: core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy [hunk 2]
```diff
@@ -73,9 +81,21 @@ class TestBintrayRelease {
...
+        uploadTask.packageName = "test"
+        uploadTask.executeWithoutThrowingTaskFailure()
 
-        project.buildDir.deleteDir()
+        assertThat(uploadTask.dryRun).isFalse()
```
Comment (friedger) [inline core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy]: This should assert for isTrue. However, the task is not configured (see comment above).

@devisnik  @hzsweers  Any ideas how to improve that?

Comment (hzsweers) [inline core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy]: I wasn't able to get this configuration up and running for some reason,
both before and after my changes. Not sure if it has something to do with
the dummy fixture directory being too lean or something else :/

Comment (friedger) [inline core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy]: this branch is working for me: ./gradlew test runs 3 tests.

Comment (friedger) [inline core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy]: @hzsweers I have moved the test for bintrayUpload into an integratiaon test using `GradleConnector` and `ProjectConnector`.

**Comment by friedger** [comment]
@blundell ready for review

File: core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy (hunk 2 above)
Comment (hzsweers) [inline core/src/test/groovy/com/novoda/gradle/release/TestConfiguration.groovy]: Nice work, didn't know about those!

**Comment by friedger** [comment]
@hzsweers @devisnik Actually, there is even better support since 2.6 https://docs.gradle.org/current/userguide/test_kit.html

**Comment by friedger** [comment]
https://github.com/novoda/bintray-release/pull/68 improves the testing of the bintrayUpload task

**Comment by hzsweers** [comment]
You may have trouble building against gradle 2.4+. I tried that as well but
kept hitting an internal exception in the bintray-release plugin

**Comment by friedger** [comment]
@hzsweers using bintray-release 0.3.4+ works for me.

**Comment by hzsweers** [comment]
Great! I'll give it a shot today

**Comment by hzsweers** [comment]
Confirmed that it's working for us now too. Thanks for getting all this resolved guys!

**Pull request merged**